test/functional/pdo_sqlsrv/pdo_batch_query.phpt

```diff
@@ -0,0 +1,202 @@
+              array(0, 31, 127, 255, 1, 10),
+              array(4534, -212, 32767, 0, 7, -32768),
+              array(-1, 546098342985600, 9223372000000000000, 5115115115115, 7, -7),
+              array(0, 1, 0, 0, 1, 1),
+             );
+
+// Column names
+$colName = array('c1_int', 'c2_tinyint', 'c3_smallint', 'c4_bigint', 'c5_bit');
+
+// Fetch one column at a time
+$expectedCols = 1;
+
+// Number of table rows
+$expectedRows = sizeof($data[0]);
+
+// Expected result sets = number of columns, since the batch fetches each column sequentially
+$expectedResultSets = sizeof($colName);
+
+function checkErrors($expectedError)
+{
+    // TODO: Fill this in once PDO error reporting is fixed
+}
+
+function checkColumnsAndRows($stmt, $cursor, $before)
+{
+    global $expectedCols, $expectedRows;
+    
+    $cols = $stmt->columnCount();
+    
+    if ($cols != $expectedCols) {
+        fatalError("Incorrect number of columns returned with $cursor cursor. Expected $expectedCols columns, got $cols columns\n");
+    }
+
+    $rows = $stmt->rowCount();
+
+    // Buffered cursors always return the correct number of rows. Other cursors
+    // return -1 rows before fetching. Static and keyset cursors return -1 even
+    // after fetching, while forward and dynamic cursors return the correct
+    // number of rows after fetching.
+    if ($cursor == 'buffered') {
+        if ($rows != $expectedRows) {
+            fatalError("Incorrect number of columns returned with buffered cursor. Expected $expectedRows rows, got $rows rows\n");
+        }
+    } else {
+        if ($before) {
+            if ($rows !== -1) {
+                fatalError("Incorrect number of rows returned before fetching with a $cursor cursor. Expected -1 rows, got $rows rows\n");
+            }
+        } else {
+            if ($cursor == 'static' or $cursor == 'keyset') {
+                if ($rows !== -1) {
+                    fatalError("Incorrect number of rows returned before fetching with a $cursor cursor. Expected -1 rows, got $rows rows\n");
+                }
+            } else {
+                if ($rows != $expectedRows) {
+                    fatalError("Incorrect number of columns returned with buffered cursor. Expected $expectedRows rows, got $rows rows\n");
+                }
+            }
+        }
+    }
+}
+
+function printCursor($element)
+{
+    $cursor = 'forward';
+    switch($element)
+    {
+        case 0:
+        echo "Testing with forward cursor...\n";
+        break;
+        case 1:
+        echo "Testing with dynamic cursor...\n";
+        $cursor = 'dynamic';
+        break;
+        case 2:
+        echo "Testing with static cursor...\n";
+        $cursor = 'static';
+        break;
+        case 3:
+        echo "Testing with keyset cursor...\n";
+        $cursor = 'keyset';
+        break;
+        case 4:
+        echo "Testing with buffered cursor...\n";
+        $cursor = 'buffered';
+        break;
+    }
+    return $cursor;
+}
```

Comment:
Please use a default case for switch statement for a catch all. Also, please use PDO cursor types instead of `$element`.

Reply:
The cursor array is an array of arrays (because we have to specify `PDO::SQLSRV_ATTR_CURSOR_SCROLL_TYPE` if `PDO::ATTR_CURSOR` is `PDO::CURSOR_SCROLL`). I could use a nested switch for the case where `PDO::ATTR_CURSOR` is `PDO::CURSOR_SCROLL`?

Reply:
Can you somehow use only `PDO::SQLSRV_ATTR_CURSOR_SCROLL_TYPE`?